src/test/java/mocks/TestArtifact.java

```diff
@@ -0,0 +1,232 @@
+package mocks;
+
+import org.apache.maven.artifact.Artifact;
+import org.apache.maven.artifact.handler.ArtifactHandler;
+import org.apache.maven.artifact.handler.DefaultArtifactHandler;
+import org.apache.maven.artifact.metadata.ArtifactMetadata;
+import org.apache.maven.artifact.repository.ArtifactRepository;
+import org.apache.maven.artifact.resolver.filter.ArtifactFilter;
+import org.apache.maven.artifact.versioning.ArtifactVersion;
+import org.apache.maven.artifact.versioning.OverConstrainedVersionException;
+import org.apache.maven.artifact.versioning.VersionRange;
+
+import java.io.File;
+import java.util.Collection;
+import java.util.List;
+
+public class TestArtifact implements Artifact {
```

Comment:
this stub and the other TestProjectBuilder , would it be possible to drop them and use e.g. Mockito?

Reply:
It would be difficult to use Mockito for these test doubles. I attempted to originally but because these kind of sit inside maven guts it's hard to control the behavior tell. This is actually less code.